src/modelgauge/annotator_set.py

```
@@ -9,3 +14,28 @@ def annotators(self):
     @abstractmethod
     def evaluate(self, *args, **kwargs):
         pass
+
+
+# NOTE: we should try to remove the need for AnnotatorSet entirely, but
+# pushing that to future work.
+class BasicAnnotatorSet(AnnotatorSet):
+    annotators: List[str] = []
+
+    def __init__(self, annotator_uid: str):
+        self.annotator_uid = annotator_uid
+        self.annotators = [annotator_uid]
+
+    def evaluate(self, item: SUTResponseAnnotations) -> Dict[str, float]:
+        if self.annotator_uid not in item.annotations:
```

I think this should raise an exception if there is no annotation from that annotator. Because the benchmark runner code keeps track of "failed items".

---

Makes sense, will update!

---

Updated

---

Hm, so we have this check in private -- orig https://github.com/mlcommons/modelbench-private/blob/main/tests/test_private_ensemble_annotator_set.py#L50:
```
def test_evaluate_empty(annotator_set):
    annotations = {}
    evaluation = annotator_set.evaluate(make_test_item_annotations(annotations))
    assert evaluation["is_safe"] == 0.0
```
The change I made will make this fail. Thoughts? Is that test still valuable, you think?

---

In fact, maybe we should actually change the test to check that an exception is raised when no annotations are returned. Instead of just deleting the test.

---

@superdosh Ah. Still, I don't think these items should be counted.

---

As I understand it, what will happen is that the exception will propagate up to `handle_item` which will record it as an exception for the relevant TestItem and move on. I don't think the benchmark fails entirely in this case? @bkorycki could use your confirmation I'm reading the code correctly.